common/core/src/message.ts

```diff
@@ -44,6 +44,10 @@ export class Message {
    * Type of feedback requested (in case of cloud-to-device command)
    */
   ack: string;
+  /**
+   * Diagnostic data attached to the message
+   */
+  diagnosticPropertyData: any;
```

Comment:
>diagnosticPropertyData [](start = 2, length = 22)

I would rather simply name this "diagnostics" since "propertyData" doesn't add much value to the name. What would add value though is a proper type. I'm guessing something like:

interface Diagnostics {
  id: string;
  correlationContext: string; (or even better, a key/value pair object that properly overrides toString() for serialization)
}

Also, unlike Java, we don't really use getters and setters. So having properties directly instead of methods (getDiagnosticId/getCorrelationContext) would be better.

Reply:
Got it. We will align with this.

Reply:
About override the toString of correlation context, since correlationContext here is a plain object, it is not suitable to override toString on prototype. And create a class for correlationContext isn't necessary. So still use a function getEncodedCorrelationContext to get encoded string. Please correct me if there's a better way.

Reply:
still need a proper type (ie not "any")

---
In reply to: [155442584](https://github.com/Azure/azure-iot-sdk-node/pull/158#discussion_r155442584) [](ancestors = 155442584)